app/controllers/posts_controller.rb

```diff
@@ -1,13 +1,43 @@
 class PostsController < ApplicationController
-  before_action :find_user, only: %i[index show]
-  before_action :find_post, only: [:show]
+  before_action :find_user, only: %i[index show like unlike]
+  before_action :find_post, only: %i[show like unlike]
 
   def index
     @posts = @user.posts
   end
 
   def show; end
 
+  def new
+    @user = current_user
+    @post = @user.posts.new
+  end
+
+  def create
+    @post = current_user.posts.new(post_params)
+    if @post.save
+      flash[:notice] = 'Post created successfully.'
+      redirect_to user_path(current_user)
+    else
+      render 'new'
+    end
+  end
+
+  def like
+    @like = @post.likes.new
+    @like.author = current_user
+    @like.save
+    redirect_to user_post_path(@user, @post)
+  end
+
+  def unlike
+    @like = @post.likes.find_by(post: @post) # Find the like
+    @like&.destroy # Destroy the like if found
+    redirect_to user_post_path(@user, @post)
+  end
```

Comment:
[OPTIONAL] It would be best if you implemented the following suggestions to improve your project.

1. To enhance the user experience, kindly make sure that the Post card  is clickable on the User show page so that navigating to the Post show page is easier. I can see that you implemented this on the Posts index page, kindly do the same for the posts on the User show page too.

2. It would be best if you followed the Rails convention by making sure that there is a `likes_controller.rb` that handles all the likes actions. It is not good practice to mix actions from two different entities in the same controller.

- The likes controller should have a `create` action.
- There should be a route in the config/routes.rb file for creating like (`:create`).

3. The `unlike` button doesn't work. The `likes_counter` remains the same even after the button is clicked.